ui/components/app/snaps/snap-permission-adapter/snap-permission-adapter.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import SnapPermissionCell from '../snap-permission-cell';
+
+export default function SnapPermissionAdapter({
+  snapId,
+  permissions,
+  showOptions,
+  targetSubjectsMetadata,
+  revoked,
+  approved,
+}) {
+  return permissions.map((permission, index) => (
```

Comment:
Does the adapter still serve a purpose now that we have moved the logic around?

Reply:
I think it still has the purpose. Especially in the update-list component, where multiple mapping is required for each of the permission sets. 

I wanted to avoid having to map it everywhere, looks better somehow if it's wrapped in into component. I'm also not a fan of using similar kind of logical processing or mapping inside the JSX template, it gets more difficult to understand what's going on and maintain. And, I also wanted to avoid having to add more utility functions to the top level of the component, which I would prefer doing over having the mapping inside the template, etc.

I think it still serve the purpose here 🤔